src/coingecko_pricing.py

```diff
@@ -53,30 +54,52 @@ def get_api_price(
         return None
 
 
+def price_retrievable(web3: Web3, block_start_timestamp: int) -> bool:
```

Comment:
Since you want to return `True` if price is NOT retrievable, the name should probably be changed.